enterprise/com/src/main/java/org/neo4j/com/storecopy/StoreCopyServer.java

```diff
@@ -154,13 +159,25 @@ public RequestContext flushStoresAndStreamStoreFiles( String triggerName, StoreW
                 while ( files.hasNext() )
                 {
                     File file = files.next();
-                    try ( StoreChannel fileChannel = fileSystem.open( file, "r" ) )
+
+                    // Read from paged file if mapping exists. Otherwise read through file system.
+                    final Optional<PagedFile> optionalPagedFile = pageCache.getExistingMapping( file );
+                    try ( ReadableByteChannel fileChannel = optionalPagedFile.isPresent() ?
+                                                            optionalPagedFile.get().openReadableByteChannel() :
+                                                            fileSystem.open( file, "r" ) )
                     {
                         monitor.startStreamingStoreFile( file );
                         writer.write( relativePath( storeDirectory, file ), fileChannel,
                                 temporaryBuffer, file.length() > 0 );
                         monitor.finishStreamingStoreFile( file );
                     }
+                    finally
+                    {
+                        if ( optionalPagedFile.isPresent() )
+                        {
+                            optionalPagedFile.get().close();
```

Comment:
`close()` throws `IOException` that we will catch and re-throw as a `ServerFailureException` later.
Can we keep this behaviour with `optional.map(...)`?